src/lib/mina/v1/zkapp.ts

```diff
@@ -634,6 +634,16 @@ class SmartContract extends SmartContractBase {
     return computeMaxProofsVerified(Object.values(methodData).map((d) => d.proofs.length));
   }
 
+  /**
+   * Manually set the verificaiton key.
+   */
+  static setVerificationKeyUnsafe(verificationKey: { data: string; hash: Field | string }) {
```

Comment:
I don't think the type `string` belongs in our source code, but I don't need to die on the hill right now.

There is a type `Base64VerificationKeyString` which is an alias of string at the moment, but I prefer to use it in cases where that's what we actually mean.  If we ever scope down what is allowed in this type, it is nicer to already have a semantic alias for what we mean.

Also `static _verificationKey?: { data: string; hash: Field };` is already used in this file.  Why do we allow a Field or string in this method param (I noticed that there is no type error by just removing `| string`, so is there a downstream issue)?  I'd rather see a proper `VerificationKey` type used in both places.

Reply:
What is the intended usage of `SmartContract.setVerificationKeyUnsafe`?  It seems like it should be an instance method, not a class method right?  We want to set the vk on an instance of a smart contract.

There is a relevant test at `src/examples/zkapps/dex/upgradability.ts` - could you add a case there maybe to show the usage of this?

Reply:
Oh, I see the intended use from the issue!

Reply:
It's "fine" but it's not taking full advantage of typescript to telegraph how to use the API.  When I say we shouldn't use `string`, I don't mean we shouldn't accept string inputs into the method.  I mean that it's an overly broad type annotation.  This method does not actually accept a string.  It accepts a string representation of a base64 encoded ocaml struct, and string representation of a Poseidon hash.  We have an opportunity to communicate that to the developer explicitly with types, and we're not taking advantage of it.

Like I said, I don't need to die on this hill for this PR, but as an on-boarding exercise, I do think there could be room for improving the clarity argument types.